src/ir_Sharp.h

```diff
@@ -38,8 +38,10 @@ const uint8_t kSharpAcFanHigh = 0b101;  // 5 (FAN3)
 const uint8_t kSharpAcFanMax =  0b111;  // 7 (FAN4)
 const uint8_t kSharpAcByteTemp = 4;
 const uint8_t kSharpAcBytePower = 5;
-const uint8_t kSharpAcBitPowerOffset = 4;
-const uint8_t kSharpAcBitModeNonAutoOffset = 5;  // 0b00100000
+const uint8_t kSharpAcBitPowerOffset = 4;  // 0b00xx0000
```

Comment:
So it looks like this is similar to some other A/Cs that use a "power toggle/is changing" bit.
e.g.
`0b000x0000` is the desired power state bit (i.e. power on or power off) e.g. `0` is off & `1` is on.
and
`0b00x00000` is the power toggle (the power setting is being changed) bit. `1` is the power state changed, `0` the power state hasn't changed?

Does that match up with your findings?
Also, can I have some `state[]` lines for these cases plus some descriptions from you saying what each state means/does/is doing/what was pressed on the remote.

Reply:
@crankyoldgit Thank you for helping me with this PR. I really appreciate your willingness to teach me :) I'll try to stick with the guides.

Oh I got it. As your suggestion, I think it is better with this
```suggestion
const uint8_t kSharpAcBitPowerOffset = 4;  // 0b000x0000
const uint8_t kSharpAcBitPreviousPowerOffset = 5;  // 0b00x00000
```
Based on my test, the data I got was:

```
Command ON (previously OFF) -> 0xAA 5A CF 10 CB 11 22 00 08 80 00 E0 51
Command ON (previously ON)  -> 0xAA 5A CF 10 CB 31 22 00 08 80 04 E0 31
Command OFF (previously ON) -> 0xAA 5A CF 10 CB 21 22 00 08 80 00 E0 61
```
For the command OFF (previously OFF), I can't reproduce this scenario since I can't make changes when it's OFF. Note: The data above was obtained with the other parameters constant.

Reply:
Can I get you to do a little more data collection/testing please?

Can I get you to try (and capture each):
1. Start with the AC off.
2. Press Command ON. (Prev should be OFF etc)
3. Press Command ON again. (prev should be ON)
4. Some other command e.g. change temp etc. (What does `state[5]` look like etc?) i.e. What is Power and Prev?
5. Press Command OFF.

Basically, like the data you've collected but with an extra step in the middle.

Also. ByteManual (aka. `state[10]`) .. I'm wondering if it contains a value that corresponds to what physical button was pressed on the remote. e.g. Fan button = 5, Mode button = 0, Temperature button = 4.  Some remotes/AC protocols do this. It might explain what it's for better. You should be able to confirm this pretty easily by seeing if changes with different buttons etc.

Reply:
FYI, I've pushed a change to your branch. You'll need to either merge, or do a `git pull` in your branch before you change things etc to re-sync. See https://github.com/crankyoldgit/IRremoteESP8266/pull/1074/commits/ddd28b909c447689c4a68006c6dd721937400104

Reply:
How should I present the collected / testing data? 

> ```
> Command ON (previously OFF) -> 0xAA 5A CF 10 CB 11 22 00 08 80 00 E0 51
> Command ON (previously ON)  -> 0xAA 5A CF 10 CB 31 22 00 08 80 04 E0 31
> Command OFF (previously ON) -> 0xAA 5A CF 10 CB 21 22 00 08 80 00 E0 61
> ```

Would this be acceptable?

Reply:
That will work. However the `state[]` line from `IRrecvDumpV2` is best. Less work for me to copy 
 & convert it into an array for the testing code. ;-)
e.g. https://github.com/juliussin/IRremoteESP8266/blob/update-sharp/test/ir_Sharp_test.cpp#L755-L759

Reply:
Thanks for the extra data, I'll take a look in the morning. I'm back to being knee deep in HitachiAc1 code at moment, and its late/early morning here. ;-)

Any luck seeing if `state[10]` is linked to which button was pressed?

Reply:
> 
> 
> That will work. However the `state[]` line from `IRrecvDumpV2` is best. Less work for me to copy
> & convert it into an array for the testing code. ;-)
> e.g. https://github.com/juliussin/IRremoteESP8266/blob/update-sharp/test/ir_Sharp_test.cpp#L755-L759

Sorry I'm not quiet understand how the test code works, so I let the data there in variable collect1 until collect 8 with all the comments :)



> 
> 
> Can I get you to do a little more data collection/testing please?
> 
> Can I get you to try (and capture each):
> 
>     1. Start with the AC off.
> 
>     2. Press Command ON. (Prev should be OFF etc)
> 
>     3. Press Command ON again. (prev should be ON)
> 
>     4. Some other command e.g. change temp etc. (What does `state[5]` look like etc?) i.e. What is Power and Prev?
> 
>     5. Press Command OFF.
> 
> 
> Basically, like the data you've collected but with an extra step in the middle.
> 
> Also. ByteManual (aka. `state[10]`) .. I'm wondering if it contains a value that corresponds to what physical button was pressed on the remote. e.g. Fan button = 5, Mode button = 0, Temperature button = 4. Some remotes/AC protocols do this. It might explain what it's for better. You should be able to confirm this pretty easily by seeing if changes with different buttons etc.

From the collected data, we can see the pattern of ManualByte `state[10]`, its value is `0x00` if the `Power` or `Mode` is changing from the previous state. `0x04` if the `Temp` is changing. `0x05` if the `Fan` is changing. It is based on my remote. But in my DIY Remote, I set it's value to `0x00` for all previous state and no error occurred.

 

> 
> 
> Thanks for the extra data, I'll take a look in the morning. I'm back to being knee deep in HitachiAc1 code at moment, and its late/early morning here. ;-)
> 
> Any luck seeing if `state[10]` is linked to which button was pressed?

 Sure, thank you for all your help :DD All of this is important knowledge and learning for me. I appreciate your time to review this PR.
No need to rush, lots of free time in the middle of this quarantine. :)